docs/topics/sealed-classes.md

```diff
@@ -1,22 +1,26 @@
 [//]: # (title: Sealed classes and interfaces)
 
 _Sealed_ classes and interfaces represent restricted class hierarchies that provide more control over inheritance.
+Sealed classes and interfaces, in contrast to `internal` classes, ensure inherent exhaustiveness in `when` expressions within their declared module or package.
 All direct subclasses of a sealed class are known at compile time. No other subclasses may appear outside
-the module and package within which the sealed class is defined. For example, third-party clients can't extend your sealed class in their code.
-Thus, each instance of a sealed class has a type from a limited set that is known when this class is compiled.
+the module and package within which the sealed class is defined. 
```

Comment:
Actually, I'm not sure about having to be in the same package (but definitely they are all in the same module).